lib/presentation/mixins/play_video_action_mixin.dart

```diff
@@ -1,20 +1,26 @@
+import 'package:fluffychat/utils/interactive_viewer_gallery.dart';
 import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/widgets/hero_page_route.dart';
 import 'package:fluffychat/widgets/video_viewer_desktop_theme.dart';
 import 'package:fluffychat/widgets/video_viewer_mobile_theme.dart';
 import 'package:flutter/material.dart';
 
 mixin PlayVideoActionMixin {
-  void playVideoAction(BuildContext context, String uriOrFilePath) async {
-    await showDialog(
-      context: context,
-      useRootNavigator: PlatformInfos.isWeb,
-      useSafeArea: false,
-      builder: (context) {
-        if (PlatformInfos.isWeb || PlatformInfos.isDesktop) {
-          return VideoViewerDesktopTheme(path: uriOrFilePath);
-        }
-        return VideoViewerMobileTheme(path: uriOrFilePath);
-      },
+  void playVideoAction(
+    BuildContext context,
+    String uriOrFilePath, {
+    String? eventId,
+  }) async {
+    Navigator.of(context, rootNavigator: PlatformInfos.isWeb).push(
+      HeroPageRoute(
+        builder: (context) {
+          return InteractiveViewerGallery(
+            itemBuilder: PlatformInfos.isWeb
+                ? VideoViewerDesktopTheme(path: uriOrFilePath)
+                : VideoViewerMobileTheme(path: uriOrFilePath),
```

Comment:
should check mobile instead, then desktop can work in future
```suggestion
                PlatformInfos.isMobile
                ? VideoViewerMobileTheme(path: uriOrFilePath)
                : VideoViewerDesktopTheme(path: uriOrFilePath),
```

Reply:
done